frontend/src/pages/new_question/nodeSelector/NodeSelector.jsx

```diff
@@ -0,0 +1,204 @@
+  const [open, toggleOpen] = useState(false);
+  const [options, setOptions] = useState([]);
+  const displayAlert = useContext(AlertContext);
+  const { concepts } = useContext(BiolinkContext);
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+
+  /**
+   * Get dropdown options for node selector
+   *
+   * Uses options:
+   * - includeExistingNodes
+   * - includeCategories
+   * - includeCuries
+   *
+   * Sets new dropdown options
+   */
+  async function getOptions() {
+    toggleLoading(true);
+    const newOptions = original ? [] : [{ name: 'Turn into new term', key: null }];
+    // allow user to select an existing node
+    if (includeExistingNodes) {
+      newOptions.push(...existingNodes);
+    }
+    // add general concepts to options
+    if (includeCategories) {
+      const includedCategories = concepts.filter(
+        (category) => category.toLowerCase().includes(debouncedSearchTerm.toLowerCase()),
+      ).flatMap((category) => (
+        [
+          {
+            category: [category],
+            name: strings.displayCategory(category),
+          },
+          {
+            category: [category],
+            name: strings.setify(category),
+            is_set: true,
+          },
+        ]
+      ));
+      newOptions.push(...includedCategories);
+    }
+    // fetch matching curies from external services
+    if (includeCuries && debouncedSearchTerm.length > 3) {
+      if (debouncedSearchTerm.includes(':')) { // user is typing a specific curie
+        newOptions.push({ name: debouncedSearchTerm, id: debouncedSearchTerm });
+      } else {
+        const fetchedCuries = await fetchCuries(debouncedSearchTerm, displayAlert);
+        newOptions.push(...fetchedCuries);
+      }
+    }
+    toggleLoading(false);
+    setOptions(newOptions);
+  }
+
+  useEffect(() => {
+    if (open) {
+      getOptions();
+    } else {
+      setOptions([]);
+    }
+  }, [open, debouncedSearchTerm]);
+
+  /**
+   * Create a human-readable label for every option
+   * @param {object} opt autocomplete option
+   * @returns {string} Label to display
+   */
+  function getOptionLabel(opt) {
+    let label = '';
+    if (opt.key) {
+      label += `${opt.key}: `;
+    }
+    if (opt.name) {
+      return label + opt.name;
+    }
+    if (opt.id && Array.isArray(opt.id) && opt.id.length) {
+      return label + opt.id.join(', ');
+    }
+    if (opt.category && Array.isArray(opt.category)) {
+      if (opt.category.length) {
+        return label + opt.category.join(', ');
+      }
+      return `${label} Any`;
+    }
+    return '';
+  }
+
+  /**
+   * Update query graph based on option selected
+   * @param {*} e click event
+   * @param {object|null} v value of selected option
+   */
+  function handleUpdate(e, v) {
+    // reset search term back when user selects something
+    updateSearchTerm('');
+    if (v && 'key' in v) {
+      // key will only be in v when switching to existing node
+      changeNode(v.key);
```

Comment:
See 2a004e7. I like "reference" because it implies that it is not the "original". I don't get the same from "pointer".

Reply:
"Reference" is fine. If "reference" and "pointer" are not already tied tightly together in your brain, you haven't spent enough time writing C. I think I'd prefer "setReference" to "changeReference" because this works even when the node does not have a reference to begin with. One can re-"set" that which is already set, but one cannot "change" that which does not exist.

Reply:
Changed in 12a7b2e5f66bb2314da300373b739b96d9053778

Reply:
Changed in 12a7b2e5f66bb2314da300373b739b96d9053778